doc/langdef.md

```diff
@@ -997,12 +1011,16 @@ functions and operators.
 
 ### Extension Functions
 
-It is possible to add extension functions to CEL, which then behave in no way
-different than standard functions. The mechanism how to do this is
-implementation dependent and usually highly curated. For example, an application
-domain of CEL can add a new overload to the `size` function above, provided this
-overload's argument types do not overlap with any existing overload. For
-methodological reasons, CEL disallows to add overloads to operators.
+It is possible to add extension functions to CEL, which then behave consistently
+with standard functions. The mechanism for doing this is implementation
+dependent and usually highly curated. For example, an application domain of CEL
+can add a new overload to the `size` function above, provided this overload's
+argument types do not overlap with any existing overload. For methodological
+reasons, CEL does not allow overloading operators.
+
+Like standard functions, extension functions must be free from observable side
+effects in order to prevent expressions from having undefined results, since CEL
+does not guarantee evaluation order of sub-expressions.
```

Comment:
Clarifying this here so we can remove qualifying language later on. (Though there are already _other_ places that do not have qualifying language and simply assert that CEL expressions are free of side effects.)